src/nplinker/genomics/antismash/downloader.py

```diff
@@ -0,0 +1,85 @@
+import os
+import shutil
+from nplinker.logconfig import LogConfig
+from nplinker.utils import download_and_extract_archive
+from nplinker.utils import list_dirs
+from nplinker.utils import list_files
+
+
+logger = LogConfig.getLogger(__name__)
+
+# urls to be given to download antismash data
+ANTISMASH_DB_PAGE_URL = 'https://antismash-db.secondarymetabolites.org/output/{}/'
+ANTISMASH_DB_DOWNLOAD_URL = 'https://antismash-db.secondarymetabolites.org/output/{}/{}'
+
+# The antiSMASH DBV2 is for the availability of the old version, better to keep it.
+ANTISMASH_DBV2_PAGE_URL = 'https://antismash-dbv2.secondarymetabolites.org/output/{}/'
+ANTISMASH_DBV2_DOWNLOAD_URL = 'https://antismash-dbv2.secondarymetabolites.org/output/{}/{}'
+
+
+def _check_roots(download_root: str, extract_root: str):
+    if download_root == extract_root:
+        raise ValueError(
+            "Identical path of download directory and extract directory")
+
+
+def _check_extract_path(extract_path: str):
+    if os.path.exists(extract_path):
+        # check if extract_path is empty
+        files = list(os.listdir(extract_path))
+        if len(files) != 0:
+            raise ValueError(f'Nonempty directory: "{extract_path}"')
+    else:
+        os.makedirs(extract_path, exist_ok=True)
+
+
+def download_and_extract_antismash_metadata(refseq_assembly_id: str,
```

Comment:
```suggestion
def download_and_extract_antismash_data(antismash_id: str,
```
`metadata` does not make sense for antismash, since it has no metatdata. This function handles with antismash BGC data.

I feel it's better to rename `refseq_assembly_id` to `antismash_id`, because for a new user the `refseq_assembly_id` is quite confusing and looks unrelated to antismash. But in the docstring, it's necessary to mention the current `antismash_id` is actually `refseq_assembly_id` (see next comment in docstring).

Note that remember to change all the other places that uses the changed names.